mvpa2/tests/test_clf.py

```diff
@@ -1093,7 +1093,40 @@ def test_gideon_weird_case(self):
             xx = pl.hist(accs, bins=bins, align='left')
             pl.xlim((0. - step/2, 1.+step/2))
 
+    @sweepargs(clf=clfswh['multiclass'])
+    def test_diff_len_labels_str(self, clf):
+        # check if the classifier can handle a dataset with labels as string of
+        # variable length
+        # was failing on TreeClassifier due to np.str dtype being assumed from first
+        # returned value
+        ds = datasets['uni4small'].copy()
+        newlabels = dict([(l,l+'_'*li) for li,l in enumerate(ds.uniquetargets)])
+        ds.targets = [newlabels[l] for l in ds.targets]
 
+        clf2 = clf.clone()
+        clf2.train(ds)
+        predictions = clf2.predict(ds)
+        # predictions on the same ds as training should give same labels
+        assert(np.all(np.unique(predictions) == ds.uniquetargets))
```

Comment:
generally it is not really has to be true.  it is ok to have training error, otherwise we would be overfitting.  if there is training error, we might miss one label completely among predictions -- it is unlikely but possible.  or am I missing smth?